### PR TITLE
daemon/update_test.go: Add TestReconcilable

### PR DESCRIPTION
**Pull request by ashcrow**
```
Running tool: /usr/bin/go test -timeout 30s -run ^TestReconcilable$

PASS
ok  	github.com/openshift/machine-config-operator/pkg/daemon	0.008s
Success: Tests passed.
```

Kind of part of #101 

/cc @sdemos @kikisdeliveryservice @jlebon

### CODE REVIEW

File: pkg/daemon/update_test.go [hunk 1]
```diff
@@ -60,3 +62,162 @@ func TestUpdateOS(t *testing.T) {
...
+		},
+	}
+
+	// Verify Ignition version changes react as expected
+	isReconcilable, err := d.reconcilable(oldConfig, newConfig)
+	if err != nil {
+		t.Errorf("Expected no error. Got %s.", err)
+	}
+	if isReconcilable != false {
+		t.Errorf("Expected different ignition versions would cause irreconcilable. Recieved reconcilable.")
```
Comment (jlebon) [inline pkg/daemon/update_test.go]: Minor: s/Recieved/Received/ (and below)

File: pkg/daemon/update_test.go [hunk 2]
```diff
@@ -60,3 +62,162 @@ func TestUpdateOS(t *testing.T) {
...
+	}
+
+	// Match ignition versions
+	oldConfig.Spec.Config.Ignition.Version = "1.0"
+	isReconcilable, err = d.reconcilable(oldConfig, newConfig)
+	if err != nil {
+		t.Errorf("Expected no error. Got %s.", err)
+	}
+	if isReconcilable != true {
+		t.Errorf("Expected the same ignition version would cause reconsilable. Recieved irreconcilable.")
```
Comment (jlebon) [inline pkg/daemon/update_test.go]: Minor: s/reconsilable/reconcilable/

**Comment by ashcrow** [comment]
@jlebon It's possible to do some magic with [reflect](https://golang.org/pkg/reflect/), etc.. which I'll take a quick look at. I do agree it would be nice to drop some of the repetitiveness, but I am wary of making the tests hard to read or too magical.

**Comment by jlebon** [comment]
Ahh much better like this!
/lgtm

**Pull request merged**